backend/main/src/Group/Domain/Entities/Container.ts

```diff
@@ -0,0 +1,99 @@
+import { Food } from "src/Group/Domain/Entities/Food";
+import { AggregateRoot, Err, Identifier, Ok, Result } from "src/Shared";
+import { DomainObjectError } from "src/Shared";
+
+interface IContainerProps {
+	name: string;
+	foods: Food[];
+}
+
+/**
+ * This class is container class.
+ * The name of container must be shorter than equal to 50 and greater than equal 1.
+ */
+export class Container extends AggregateRoot<string, IContainerProps> {
+	static create(
+		id: ContainerId,
+		props: IContainerProps,
+	): Result<Container, ContainerDomainError> {
+		if (props.name.length < 1) {
+			return Err(new ContainerDomainError("Container name is too short"));
+		}
+		if (props.name.length > 50) {
+			return Err(new ContainerDomainError("Container name is too long"));
+		}
+		return Ok(
+			new Container(id, {
+				...props,
+			}),
+		);
+	}
+
+	get name(): string {
+		return this.props.name;
+	}
+
+	get foods(): Food[] {
+		return this.props.foods;
+	}
+
+	/**
+	 * Change the name of container.
+	 * @param name
+	 */
+	public changeName(name: string): Result<Container, ContainerDomainError> {
+		return Container.create(this.id, {
+			...this.props,
+			name: name,
+		});
+	}
+
+	/**
+	 * Add food object to the container.
+	 * @param food
+	 */
+	public addFood(food: Food): Result<Container, ContainerDomainError> {
+		const foods = [...this.props.foods, food];
+		return Container.create(this.id, {
+			...this.props,
+			foods: foods,
+		});
+	}
+
+	/**
+	 * Remove food object from the container.
+	 * @param food
+	 */
+	public removeFood(food: Food): Result<Container, ContainerDomainError> {
```

Comment:
It's hard to realize that this method has a restriction.
It's better to write a comment about the conditions of error.

Reply:
ohh, thank you. I fixed it.